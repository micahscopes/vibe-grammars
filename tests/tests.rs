use dir_test::dir_test;
use insta::assert_snapshot;
use pest::Parser;
use vibe_grammars::{elements::{Parser as ElementsParser, Rule as ElementsRule}, util::to_sexpr};

#[dir_test(
    dir: "$CARGO_MANIFEST_DIR/tests/elements",
    glob: "chord*.txt",
)]
fn test_chords(fixture: dir_test::Fixture<&str>) {
    let input = *fixture.content();
    let result = ElementsParser::parse(ElementsRule::chords, input).unwrap();
    let result = to_sexpr(result, 0);
    let path = fixture.path();
    use std::path::Path;
    let path = Path::new(&path);
    let stem = path.file_stem().unwrap().to_str().unwrap();
    println!("Stem: {}", stem);
    assert_snapshot!(stem, result);
}