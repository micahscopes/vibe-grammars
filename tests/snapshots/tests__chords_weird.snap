---
source: tests/tests.rs
assertion_line: 19
expression: result
---
( chords
  `Csus4(13) Cmaj7#5/Eb`
  ( chord
    `Csus4(13)`
    ( note
      `C`
      ( base
        `C`
      )
    )
    ( extension
      `sus4`
      ( modifier
        `sus`
      )
      ( degree
        `4`
      )
    )
    ( parenthetical_extension
      `(13)`
      ( extension
        `13`
        ( degree
          `13`
        )
      )
    )
  )
  ( chord
    `Cmaj7#5/Eb`
    ( note
      `C`
      ( base
        `C`
      )
    )
    ( quality
      `maj`
    )
    ( extension
      `7`
      ( degree
        `7`
      )
    )
    ( extension
      `#5`
      ( accidental
        `#`
      )
      ( degree
        `5`
      )
    )
    ( slash_bass
      `/Eb`
      ( note
        `Eb`
        ( base
          `E`
        )
        ( accidental
          `b`
        )
      )
    )
  )
)
